src/stack/stack.jl

```diff
@@ -161,12 +165,15 @@ Base.checkbounds(T::Type, s::AbstractDimStack, I...) = checkbounds(T, CartesianI
 
 @inline Base.keys(s::AbstractDimStack{K}) where K = K
 @inline Base.propertynames(s::AbstractDimStack{K}) where K = K
-Base.setindex(s::AbstractDimStack, val::AbstractBasicDimArray, name) =
+@inline Base.setindex(s::AbstractDimStack, val::AbstractBasicDimArray, name::Symbol) =
     rebuild_from_arrays(s, Base.setindex(layers(s), val, name))
 Base.NamedTuple(s::AbstractDimStack) = NamedTuple(layers(s))
-
-# Remove these, but explain
-Base.iterate(::AbstractDimStack, args...) = error("Use iterate(layers(s)) rather than `iterate(s)`") #iterate(layers(s), args...)
+Base.collect(st::AbstractDimStack) = parent([st[D] for D in DimIndices(st)])
+Base.Array(st::AbstractDimStack) = collect(st)
+Base.vec(st::AbstractDimStack) = vec(collect(st))
+@propagate_inbounds Base.iterate(st::AbstractDimStack) = iterate(st, 1)
+@propagate_inbounds Base.iterate(st::AbstractDimStack, i) = 
+    i > length(st) ? nothing : (st[DimIndices(st)[i]], i + 1)
```

Comment:
Any way to hold off on re-instantiating the DimIndices object every time?  Indexing with linear indices (single integers) currently does what I want anyway, it just needs to be forwarded correctly?

Reply:
It should be free anyway. And if you index with linear indices it has to do DimIndices(A)[i] internally anyway!

Mixed dimension stacks indexing really works properly now because of Dimension indexing is used internally.